### ISSUE TITLE
All values are touched

### ISSUE THREAD

**Issue by smnplk** [luciodale/fork#4]
I might be misusing this lib somehow, so please steer me in the right direction. 
My update-ad handler receives all values as touched, even if i don't change any values in the form before I press the save button. 

Here is my form component

```clojure 
(defn ad-edit-form []
  (let [cars (rf/subscribe [:cars])
        ad (rf/subscribe [:selected-ad])]
      (when @ad
        [fork/form {:path :ad-edit-form
                    :form-id "ad-edit-form"
                    ;; todo: use select-keys with clojure.walk/stringify-keys here
                    :initial-values {"make" (:make @ad) "model" (:model @ad)
                                     "year" (:year @ad) "kilometers" (:kilometers @ad)}
                    :prevent-default? true
                    :clean-on-unmount? true
                    :on-submit #(rf/dispatch [:update-ad %])}

          (fn [{:keys [values form-id handle-change touched
                       submitting? handle-submit handle-blur]}]
           ;; (println touched)
            [:form.w-full {:id form-id}
              [:div.flex.flex-wrap.-mx-3.mb-6
                [:div {:class "w-full md:w-1/2 px-3 mb-6 md:mb-0 mt-2"}
                  [forms/select-field {:label "make" :value (values "make")
                                        :name "make"
                                        :options (keys @cars)
                                        :handle-change handle-change}]]
                [:div {:class "w-full md:w-1/2 px-3 mb-6 md:mb-0 mt-2"}
                  [forms/select-field {:label "model" :value (values "model" "1")
                                        :name "model"
                                        :options ["1" "2" "3"]
                                        :handle-change handle-change}]]
                [:div {:class "w-full md:w-1/2 px-3 mb-6 md:mb-0 mt-2"}
                  [forms/text-field {:label "year" :name "year" :value (values "year") :handle-change handle-change  :handle-blur handle-blur}]]
                [:div {:class "w-full md:w-1/2 px-3 mb-6 md:mb-0 mt-2"}
                  [forms/text-field {:label "kilometers" :name "kilometers" :value (values "kilometers") :handle-change handle-change :handle-blur handle-blur}]]
                [:div.flex.items-center.justify-center.w-full.mt-4
                  [:button.bg-teal-500.hover:bg-teal-300.text-white.font-bold.py-2.px-4.rounded
                    {:on-click handle-submit :disabled submitting?} "Save"]]]])])))


```

```clojure
(rf/reg-event-fx
 :update-ad
 [(fork/on-submit :ad-edit-form)]
 (fn [{db :db} [_ {:keys [values]}]]
   (js/alert values)
   (.log js/console values)
   {:db (fork/set-submitting db :ad-edit-form false)}))
```

**Comment by luciodale** [comment]
Hello there! That is in fact the intended behavior. The fields are set to `touched` after any submitting attempt to allow the errors to be displayed. 

If you want to know whether any field as been touched at all, there’s a `dirty?` key that gets passed to your submit handler along with the `values` that servers specifically that purpose.

**Comment by smnplk** [comment]
Hi, thanks for clarifying.  :) 

I added initial-values as dispatch data 
```clojure 

  :on-submit #(rf/dispatch [:update-ad % initial-values])

```

Then I grab initial-values from submit handler.

```clojure
(rf/reg-event-fx
 :update-ad
 [(fork/on-submit :ad-edit-form)]
 (fn [{db :db} [_ {:keys [values dirty?]} initial-values]]
   (println "dirty? " dirty?)
   (println "init values " initial-values)
   (println "new values " values)
   {:db (fork/set-submitting db :ad-edit-form false)}))
```
Now I can just compare initial-values  to values and  see which keys changed. The only thing that doesn't seem to work correctly in my case is the dirty? key, it returns true only if **all** the values have changed.

**Comment by luciodale** [comment]
It was indeed a bug. I fixed it in `1.2.6`. Now you can use `dirty` instead of `dirty?`, which gives you either `nil` or a map of changed values. :)

**Comment by smnplk** [comment]
Perfect. Works like a charm, thank you.
